frontend/src/hooks/useApi.ts

```diff
@@ -0,0 +1,26 @@
+import CustomAxios from 'customAxios';
+import { useDispatch } from 'react-redux';
+import { showLoading, hideLoading } from 'react-redux-loading-bar';
+
+export const useApi = () => {
+  const dispatch = useDispatch();
+  const axios = CustomAxios();
+
+  axios.interceptors.request.use(
+    config => {
+      dispatch(showLoading());
+      return config;
+    },
+    error => dispatch(hideLoading()),
```

Comment:
I see this extra logic a lot in the codebase.  You can just use a `finally` to `hideLoading()`.